File: src/main/java/com/converter/api/dto/UserForm.java
package com.converter.api.dto;

import com.converter.api.validation.UnregisteredUserEmail;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

public record UserForm(
        @NotNull @NotBlank @Email @UnregisteredUserEmail String email,
        @NotNull @NotBlank @Size(min = 8, max = 24) String password) {
}

File: src/main/java/com/converter/api/dto/UserForm.kt
package com.converter.api.dto

import com.converter.api.validation.UnregisteredUserEmail
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

data class UserForm(
    @NotNull @NotBlank @Email @UnregisteredUserEmail val email: String,
    @NotNull @Size(min = 8, max = 24) val password: String
)